### PR TITLE
[4.0] replace own joomla's highlighter with mark.js

### PR DESCRIPTION
**Pull request by dgrammatiko**
Pull Request for Issue # .

### Summary of Changes
- Remove highlighter.(min.)js
- Remove `behavior.highlighter`
- Introduce `behavior.highlight`

### Testing Instructions
Not yet testable

### Expected result
Same highlighting as before

### Documentation Changes Required
YUP

### CODE REVIEW

File: plugins/system/highlight/highlight.php [hunk 1]
```diff
@@ -72,12 +73,11 @@ public function onAfterDispatch()
...
 
 		// Activate the highlighter.
-		JHtml::_('behavior.highlighter', $cleanTerms);
+		JHtml::_('behavior.highlight', $cleanTerms);
 
 		// Adjust the component buffer.
-		$doc = JFactory::getDocument();
 		$buf = $doc->getBuffer('component');
-		$buf = '<br id="highlighter-start" />' . $buf . '<br id="highlighter-end" />';
+		$buf = '<div class="js-highlighter" >' . $buf . '</div>';
```
Comment (C-Lodder) [inline plugins/system/highlight/highlight.php]: remove space after `class="js-highlighter" `

File: media/system/js/highlight.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+ * @version     1.0
+ */
+Joomla = window.Joomla || {};
+
+(function(Joomla) {
+	document.addEventListener('DOMContentLoaded',  function() {
+
+		if (joomla.getOptions && typeof joomla.getOptions === 'function' && Joomla.getOptions('js-highlight')) {
+			var options = Joomla.getOptions('js-highlight'),
+                markOptions = {
```
Comment (C-Lodder) [inline media/system/js/highlight.js]: indentation

**Comment by wilsonge** [comment]
OK I'm not against us moving to this new library. But let's have a think about working out a method of making the old JHtml method signature work with the new library, if possible.

**Comment by dgt41** [comment]
@wilsonge it's doable, I did this one as a POC, but I can use the old PHP API, adding just an array as the first param and doing some basic checking, you get the idea...

**Comment by wilsonge** [comment]
I'm happy with moving to something Not In House but I'd rather keep the PHP API b/c from 3.x and deprecate the old API for 5 if we can

**Comment by brianteeman** [comment]
What is the problem this is solving? Does it provide anything more than we have right now?

**Comment by dgt41** [comment]
> What is the problem this is solving? Does it provide anything more than we have right now?

Two objectives:
- vanilla javascript (no dependencies, also this script is future proof, e.g. ES6)
- less code to support in-house...

**Comment by dgt41** [comment]
I will redo the PHP part so we can keep B/C here

**Pull request closed without merging**